lib/extensions.js

```diff
@@ -80,37 +80,43 @@ module.exports = function(realmConstructor, context) {
     const { DefaultNetworkTransport } = require('realm-network-transport');
     realmConstructor._networkTransport = new DefaultNetworkTransport();
 
-    Object.defineProperty(realmConstructor.Object.prototype, "toJSON", {
-        value: function () {
-            const result = {}
-            Object.keys(this).forEach(p => result[p] = this[p]);
-            Object.keys(Object.getPrototypeOf(this)).forEach(p => result[p] = this[p]);
-            return result;
+    Object.defineProperty(realmConstructor.Collection.prototype, "toJSON", {
+        value: function (_, cache = new Map()) {
```

Comment:
`_` is the key where the current element resides in its parent (an empty string in case of the root element), and is supplied as first argument by `JSON.stringify` (see [docs](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior)).

The underscore convention is used to clarify to the reader that the argument is not used, and I changed it because of a question in the v6 draft PR: https://github.com/realm/realm-js/pull/3013#discussion_r447591166
The convention is also used in [lib/notifier.js](https://github.com/realm/realm-js/blob/sa/v10/realm-tojson/lib/notifier.js#L239).

I could add a comment to the function, stating what it is?
Something like this:
```
// The first unused argument is supplied by JSON.stringify, and is the key where the current element resides in its parent
// MDN docs: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/stringify#toJSON_behavior.
```

Reply:
I prefer the discard syntax because it immediately tells the reader that we're not using that argument rather than having them read through the entire function to figure out what's happening with the `key` argument.

Reply:
@nirinchev  why would you need to "figure out what's happening with the key argument" when reading this function. @steffenagger I know why is that used but would you think that this is a good code doWork(x, (_, _, _, value)) { }.
Lets leave it as it this since it seems its a personal preference. I still think this is not readable though. As we all know **Programs are meant to be read by humans and only incidentally for computers to execute**  :)

Reply:
Well, if there's an argument in a function, you expect it to be used. If it isn't, that's likely a mistake and a reader of the code might wonder where was this argument supposed to be used. If something is not used, it shouldn't exist in the first place. In this case, we don't have control over it, so I think the right thing to do is to tell the code reader that whatever that first argument is, we don't care about it.